src/app/modules/item/task/task-xd-pr.ts

```diff
@@ -0,0 +1,293 @@
+/**
+ * Cross-domain task proxy implementation for Bebras task API in TypeScript
+ * Adapted from task-xd-pr.js from https://github.com/France-ioi/pem-platform/
+ *
+ * It depends on jschannel.
+ */
+
+import { Observable, of, throwError } from 'rxjs';
+import { delay, map, retryWhen, switchMap, take } from 'rxjs/operators';
+import { rxBuild, RxMessagingChannel } from './rxjschannel';
+
+export interface TaskParams {
+  minScore: number,
+  maxScore: number,
+  noScore: number,
+  randomSeed: number,
+  readOnly: boolean,
+  options: Object,
+}
+export type TaskParamsValue = TaskParams | Object | string | number | undefined;
+
+export interface UpdateDisplayParams {
+  height?: number | string,
+  views?: Object,
+  scrollTop?: number,
+}
+
+// TODO : actual types
+export type TaskMetaData = any;
+export type TaskView = any;
+export type TaskViews = any;
+export type TaskGrade = any;
+export type TaskResources = any;
+export type TaskDisplayData = any;
+export type TaskLog = any;
+
+function getUrlParameterByName(name : string, url : string) : string {
+  const regex = new RegExp('[\\?&]' + name + '=([^&#]*)'),
+    results = regex.exec(url);
+  return results && results[1] ? decodeURIComponent(results[1].replace(/\+/g, ' ')) : '';
+}
```

Comment:
Honestly I only kept the function from the original file without thinking, I should have checked. There is indeed a function in helpers which can be used instead of this, so I did that.